rs/index/src/posting_list/file.rs

```diff
@@ -88,87 +85,93 @@ impl FileBackedAppendablePostingListStorage {
     pub fn flush(&mut self) -> Result<()> {
         if !self.resident {
             // Flush all mmaps
+            for mmap in self.metadata_mmaps.iter_mut() {
+                mmap.flush()?;
+            }
             for mmap in self.mmaps.iter_mut() {
                 mmap.flush()?;
             }
         }
         Ok(())
     }
 
-    fn new_backing_file(&mut self) -> Result<()> {
-        self.current_backing_id += 1;
+    fn new_backing_file(&mut self, is_metadata: bool) -> Result<()> {
```

Comment:
for readability, create 2 functions:
```
fn new_metadata_backing_file(&mut self);
fn new_posting_list_backing_file(&mut self);
```

Each of them will call `self.new_backfing_file({true, false});`. This will make the code easier to read